File: Data Structures & Algorithms/binary-tree-maximum-path-sum/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public: 
    int ans = INT_MIN;
    int solve(TreeNode* root){
        if(root == NULL)return 0;

        int l = max(0 , solve(root -> left));
        int r = max(0 , solve(root -> right));

        ans = max(ans , l + r + root -> val);

        return root -> val + max(l , r);
    }
    int maxPathSum(TreeNode* root) {

        solve(root);
        return ans;
    }
};

File: Data Structures & Algorithms/depth-of-binary-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    
    int solve(TreeNode root){
        if(root == null)return 0;
        int ans = 0;

     
            ans = Math.max(ans , 1 + solve(root.left));
        
        
            ans = Math.max(ans , 1 + solve(root.right));
        
        
        return ans;
    }
    public int maxDepth(TreeNode root) {
       
        return solve(root);
        
    }
}